test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
+      const request = [
+        {
+          roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+          permissionId: '2b33268a-7ff5-4cac-a87a-6bfc4430d34c',
+        },
+      ];
+      const input: UpdateRolePermissionInput = {
+        permissions: ['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'],
+      };
+
+      // permissionRepository
+      //   .findByIds(['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'])
+      //   .resolves(permissions);
+
+      // rolePermissionRepository.create(request).returns(request);
+
+      // permissionRepository
+      //   .createQueryBuilder('permission')
+      //   .returns(permissionQueryBuilder);
+      permissionQueryBuilder
+        .leftJoinAndSelect(
+          RolePermission,
+          'rolePermission',
+          'permission.id = rolePermission.permissionId',
+        )
+        .returns(permissionQueryBuilder);
+
+      permissionQueryBuilder
+        .where('rolePermission.roleId = :roleId', {
+          roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        })
+        .returns(permissionQueryBuilder);
+
+      permissionQueryBuilder.getMany().resolves(permissions);
+
+      mockDataSource.transaction(Arg.any()).resolves(request);
+
+      const result = await roleService.updateRolePermissions(
+        'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        input,
+      );
+      expect(result).toEqual(permissions);
+    });
+
+    it('should throw exception when adding invalid permissions to a role', async () => {
+      const input: UpdateRolePermissionInput = {
+        permissions: ['3e9e78c9-3fcd-4eed-b027-62f794680b03'],
+      };
+
+      // permissionRepository
+      //   .findByIds(['3e9e78c9-3fcd-4eed-b027-62f794680b03'])
+      //   .resolves([]);
+
+      const result = roleService.updateRolePermissions(
+        'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        input,
+      );
+      await expect(result).rejects.toThrowError(
+        new PermissionNotFoundException(
+          ['3e9e78c9-3fcd-4eed-b027-62f794680b03'].toString(),
+        ),
+      );
+    });
   });
 
-  it('should return all permissions of the role', async () => {
-    const permissions: Permission[] = [
-      {
-        id: '2b33268a-7ff5-4cac-a87a-6bfc4430d34c',
-        name: 'Customers',
-        label: 'Customers',
-      },
-    ];
-    permissionRepository
-      .createQueryBuilder('permission')
-      .returns(permissionQueryBuilder);
-    permissionQueryBuilder
-      .leftJoinAndSelect(
-        RolePermission,
-        'rolePermission',
-        'permission.id = rolePermission.permissionId',
-      )
-      .returns(permissionQueryBuilder);
-    permissionQueryBuilder
-      .where('rolePermission.roleId = :roleId', {
-        roleId: 'fcd858c6-26c5-462b-8c53-4b544830dca8',
-      })
-      .returns(permissionQueryBuilder);
-    const resp = await roleService.getRolePermissions(
-      'fcd858c6-26c5-462b-8c53-4b544830dca8',
-    );
-    expect(resp).toEqual(permissions);
+  describe('getRolePermissions', () => {
```

Comment:
Test fails.

Reply:
Will fix service related tests in a separate PR.